modal/config.py

```diff
@@ -135,6 +136,32 @@ def config_set_active_profile(env: str) -> None:
     _write_user_config(_user_config)
 
 
+def _check_config() -> None:
+    num_profiles = len(_user_config)
+    num_active = sum(v.get("active", False) for v in _user_config.values())
+    if num_active > 1:
+        raise InvalidError(
+            "More than one Modal profile is active. "
+            "Please fix with `modal profile activate` or by editing your Modal config file."
+        )
+    elif num_profiles > 1 and num_active == 0 and _profile == "default":
+        # Eventually we plan to have num_profiles > 1 with num_active = 0 be an error
+        # But we want to give users time to activate one of their profiles without disruption
+        message = dedent(
+            """
+
+            WARNING:
+
+            Support for using an implicit 'default' profile is deprecated.
+            Please use `modal profile activate` to activate one of your profiles.
+            (Use `modal profile list` to see the options.)
+
+            This will become an error in a future update.
+            """
+        )
+        deprecation_warning(date(2024, 2, 6), message)
```

Comment:
I am a little sad that this is now a boring old deprecation warning rather than the nice Rich panel in my previous implementation. It feels wrong to be using rich from within the library here (rather than the CLI). But maybe we should have something that boosts the salience of our `deprecation_warning` messages?

Reply:
Should hopefully be a relatively rare warning

Reply:
🤞